apps/test/netsim/NetSimVisualization.js

```diff
@@ -84,15 +89,74 @@ describe("NetSimVisualization", function () {
   var getVizNodeByEntityID_ = function (_type, id) {
     return this.filter(function(element){
       return element instanceof NetSimVizNode &&
-          element.getCorrespondingEntityID &&
-          element.getCorrespondingEntityID() === id;
+          element.getCorrespondingEntityId &&
+          element.getCorrespondingEntityId() === id;
     })[0];
   };
 
+  function createNode(shard, name) {
+    var newNode;
+    NetSimLocalClientNode.create(shard, name, function (e, n) {
+      newNode = n;
+    });
+    assert(newNode !== undefined, "Failed to create a client.");
+    return newNode;
+  }
+
+  it("does not mismap elements across shard reset", function () {
+    // Fake NetSim
+    var netsim = new NetSim();
+    netSimVis = new NetSimVisualization(makeRootDiv(), netsim.runLoop_);
+
+    // Connect to shard
+    testShard = fakeShard();
+    var myNode = createNode(testShard, 'myNode');
```

Comment:
how about naming these oldNode1, oldNode2, oldNode3, newNode1, newNode2, or something similar so it's easier to understand which nodes should have the same IDs as each other?